core/lib/server-toast.ts

```diff
@@ -0,0 +1,84 @@
+import { cookies } from 'next/headers';
+import { ToasterProps } from 'sonner';
+import { z } from 'zod';
+import 'server-only';
+
+interface ServerToastOptions {
+  description?: string;
+  position?: ToasterProps['position'];
+}
+
+export interface ServerToastData extends ServerToastOptions {
+  message: string;
+  variant: 'success' | 'error' | 'warning' | 'info';
+}
+
+const serverToastSchema = z.object({
+  message: z.string(),
+  description: z.string().optional(),
+  variant: z.enum(['success', 'error', 'warning', 'info']),
+  position: z
+    .enum(['top-left', 'top-right', 'top-center', 'bottom-left', 'bottom-right', 'bottom-center'])
+    .optional(),
+});
+
+const TOAST_COOKIE = 'toast-notification';
+const getExpiry = () => new Date(Date.now() + 5000);
+
+function encode(data: ServerToastData): string {
+  return btoa(JSON.stringify(data));
+}
+
+function decode(data: string): ServerToastData {
+  return serverToastSchema.parse(JSON.parse(atob(data)));
+}
+
+async function setToastCookie(data: ServerToastData) {
+  const cookieStore = await cookies();
+
+  cookieStore.set(TOAST_COOKIE, encode(data), {
+    httpOnly: true,
+    secure: process.env.NODE_ENV === 'production',
+    sameSite: 'strict',
+    path: '/',
+    expires: getExpiry(),
+  });
```

Comment:
Can we please set `partitioned: true` so this behaves well in iframes (e.g., Makeswift builder)? We could also do it in the `integrations/makeswift` branch only once this is merged